piet-direct2d/src/dwrite.rs

```diff
@@ -38,25 +50,23 @@ pub struct DwriteFactory(ComPtr<IDWriteFactory>);
 unsafe impl Send for DwriteFactory {}
 
 #[derive(Clone)]
-pub struct TextFormat(ComPtr<IDWriteTextFormat>);
+pub struct TextFormat(pub(crate) ComPtr<IDWriteTextFormat>);
+
+#[derive(Clone)]
+struct FontFamily(ComPtr<IDWriteFontFamily>);
 
-/// A builder for creating new `TextFormat` objects.
+/// A cheap to clone reference to an existing family name.
 ///
-/// TODO: provide lots more capability.
-pub struct TextFormatBuilder {
-    factory: DwriteFactory,
-    size: Option<f32>,
-    family: Option<String>,
+/// We hold onto both the wide string and the rust string.
+///
+/// This type AsRefs to both `str` and `[u16]`.
```

Comment:
A bit off-topic for review, but this would actually make a cool motivating example for `AsRef` as opposed to `Borrow`, as I know that confuses a lot of people.